privacy/ccpa/policy.go

```diff
@@ -49,35 +50,44 @@ func (p Policy) Write(req *openrtb.BidRequest) error {
 	return err
 }
 
-// Validate returns an error if the CCPA regulation value does not adhere to the IAB spec.
+// Validate returns an error if the CCPA policy does not adhere to the IAB spec.
 func (p Policy) Validate() error {
 	if p.Value == "" {
 		return nil
 	}
 
-	if len(p.Value) != 4 {
-		return errors.New("request.regs.ext.us_privacy must contain 4 characters")
+	if err := ValidateConsent(p.Value); err != nil {
+		return fmt.Errorf("request.regs.ext.us_privacy %s", err.Error())
 	}
 
-	if p.Value[0] != '1' {
-		return errors.New("request.regs.ext.us_privacy must specify version 1")
+	return nil
+}
+
+// ValidateConsent returns an error if the CCPA consent string does not adhere to the IAB spec.
+func ValidateConsent(consent string) error {
```

Comment:
I separated the functionality of validating the consent string from validating the entire policy. Allows the simpler function to be used for the "sensing" required of AMP.

Reply:
I sort of disagree on this because I'd like to avoid an additional function on top of the stack at execution time and an additional function to maintain, but this doesn't diminish its functionality and the hard work behind its design.